### PR TITLE
docs: Update music.rst to clarify music.stop()

### PR DESCRIPTION
**Pull request by microbit-mark**
As per https://github.com/bbcmicrobit/micropython/issues/416 duplicate help text from Mu and add example.

### CODE REVIEW

File: docs/music.rst [hunk 1]
```diff
@@ -113,7 +113,8 @@ Functions
 
 .. py:function:: stop(pin=microbit.pin0)
 
-    Stops all music playback on a given pin.
+    Stops all music playback on a given pin. If no pin is given eg.
+    ``music.stop(pin1)``, Pin0 is assumed.
```
Comment (microbit-carlos) [inline docs/music.rst]: The second sentence is about not using arguments, but the listed example shows one, which might be a bit confusing. Could we change this to either:

```
Stops all music playback on a given pin, eg. ``music.stop(pin1)``. If no pin is given Pin0 is assumed.
```
Or
```
Stops all music playback on a given pin. If no pin is given eg.``music.stop()``, Pin0 is assumed.
```

I think I prefer the second one.

**Comment by microbit-carlos** [comment]
Thanks Mark, I've added a minor comment, let me know what you think.

File: docs/music.rst (hunk 1 above)
Comment (microbit-mark) [reply docs/music.rst]: I'm not sure the second one resolves the difficulty that David had in the original ticket though. The first one provides an explicit example. How about a combo?

`Stops all music playback on a given pin, eg. ``music.stop(pin1)``. If no pin is given, eg. ``music.stop()`` pin0 is assumed.`

Comment (microbit-carlos) [reply docs/music.rst]: Sounds good, thanks Mark!

**Pull request merged**